### PR TITLE
Task js oop

### PR DESCRIPTION
**Pull request by MitchfFirstGit**
### Object-Oriented JavaScript

[Demo](https://mitchffirstgit.github.io/frontend-nanodegree-arcade-game/) |
[Code base](https://github.com/MitchfFirstGit/frontend-nanodegree-arcade-game/blob/master/js/app.js)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/MitchfFirstGit/Object_Oriented_JavaScript/js/app.js [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+// All enemies are placed in an array
+var allEnemies = [];
+//Location all enemies on the y axis 
+var enemyLocation = [65, 145, 230]; 
+
+// Creating  new Enemies objects and placing them in an array called allEnemies
+function createEnemy(){
+    enemyLocation.forEach(function (locationY) {
+    allEnemies.push(new Enemy(0, locationY, 270));
+});
```
Comment (OleksiyRudenko) [inline submissions/MitchfFirstGit/Object_Oriented_JavaScript/js/app.js]: The fragment above could be less verbose if you benefited from [`#Array.map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) method.

Please, refactor.

**Comment by OleksiyRudenko** [comment]
I have also noticed that the playfield scales down along with screen size proportionally. You are not required to fix this but it would look nicer if you have the playfield 100% viewport wide, of course with a max threshold for bigger screens.

Anyway, if you need any help or further clarifications, please, let me know.

**Comment by zonzujiro** [comment]
Reassigned to @OleksiyRudenko because he made a review.

**Comment by OleksiyRudenko** [comment]
@MitchfFirstGit You came up with many improvements. 

Before we proceed, could you please click **Resolve conversation** on comments that you believe you have resolved completely. As for the remaining review comments, please, ask for clarifications or provide some grounds to keep things intact if any.

Please, let me know when ready to go on with code review.

Thanks.

File: submissions/MitchfFirstGit/Object_Oriented_JavaScript/js/app.js [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+    this.speed = 100 + Math.floor(Math.random() * 100);
+  }
+  //Handles collision with the Player
+  if (
+    this.player.x < this.x + this.player.playerSizeX &&
+    this.player.x + this.player.playerSizeX > this.x &&
+    this.player.y < this.y + this.player.playerSizeY &&
+    this.player.playerSizeY + this.player.y > this.y
+  ) {
+    this.player.x = this.game.edgeXField - 101;
```
Comment (OleksiyRudenko) [inline submissions/MitchfFirstGit/Object_Oriented_JavaScript/js/app.js]: Here in everywhere - `this.game.colWidth` in place of `101` and `this.game.rowHeight` in place of `83`?

Constants make your code highly maintainable.

File: submissions/MitchfFirstGit/Object_Oriented_JavaScript/js/app.js [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+    this.score = 0;
+    document.querySelector(".helth").textContent = `Helth: ${this.helth}`;
+    document.querySelector(".score").textContent = `Score: ${this.score}`;
+    this.game.setGameToNewLevel();
+  }
+};
+
+//update Player's helth
+Player.prototype.updateHelth = function(decrement) {
+  decrement === -1 ? this.helth-- : this.helth++;
```
Comment (OleksiyRudenko) [inline submissions/MitchfFirstGit/Object_Oriented_JavaScript/js/app.js]: This construction is completely (almost) equivalent to `this.helth += decrement`.

Is there a good reason not to use this construction?

**Pull request merged**